subsys/settings/src/settings_line.c

```diff
@@ -474,7 +474,7 @@ static int settings_line_cmp(char const *val, size_t val_len,
 	size_t len_read, exp_len;
 	size_t rem;
 	char buf[16];
-	int rc;
+	int rc = 0;
```

Comment:
not really an issue as well, this function is never called for 0-length
`if (val_len == 0) {` anyway ensures rc assignment in the while